web/src/js/ducks/flows.js

```diff
@@ -1,114 +1,252 @@
-import makeList from "./utils/list"
-import Filt from "../filt/filt"
-import {updateViewFilter, updateViewList, updateViewSort} from "./utils/view"
-import {reverseString} from "../utils.js";
-import * as columns from "../components/FlowTable/FlowColumns";
+import { fetchApi as fetch } from '../utils'
+import { CMD_RESET as WS_CMD_RESET } from './websocket'
+import reduceList, * as listActions from './utils/list'
 
-export const UPDATE_FLOWS = "UPDATE_FLOWS"
-export const SET_FILTER = "SET_FLOW_FILTER"
-export const SET_HIGHLIGHT = "SET_FLOW_HIGHLIGHT"
-export const SET_SORT = "SET_FLOW_SORT"
-export const SELECT_FLOW = "SELECT_FLOW"
-
-const {
-    reduceList,
-    updateList,
-    fetchList,
-} = makeList(UPDATE_FLOWS, "/flows")
+export const WS_MSG_TYPE ='UPDATE_FLOWS'
 
+export const UPDATE_FILTER = 'FLOWS_UPDATE_FLOW_FILTER'
+export const UPDATE_HIGHLIGHT = 'FLOWS_UPDATE_FLOW_HIGHLIGHT'
+export const UPDATE_SORT = 'FLOWS_UPDATE_FLOW_SORT'
+export const WS_MSG = 'FLOWS_WS_MSG'
+export const SELECT_FLOW = 'FLOWS_SELECT_FLOW'
+export const REQUEST_ACTION = 'FLOWS_REQUEST_ACTION'
+export const REQUEST = 'FLOWS_REQUEST'
+export const RECEIVE = 'FLOWS_RECEIVE'
+export const FETCH_ERROR = 'FLOWS_FETCH_ERROR'
 
 const defaultState = {
-    all: reduceList(),
     selected: [],
-    view: [],
-    filter: undefined,
-    highlight: undefined,
-    sort: {sortColumn: undefined, sortDesc: false},
-}
-
-function makeFilterFn(filter) {
-    return filter ? Filt.parse(filter) : () => true;
+    sorter: {},
+    filter: null,
+    highlight: null,
+    list: reduceList(undefined, { type: Symbol('FLOWS_INIT_LIST') }),
 }
 
+export default function reduce(state = defaultState, action) {
+    switch (action.type) {
 
-function makeSortFn(sort){
-    let column = columns[sort.sortColumn];
-    if (!column) return;
+        case UPDATE_FILTER:
+            return {
+                ...state,
+                filter: action.filter,
+                list: reduceList(state.list, listActions.updateFilter(makeFilterFun(action.filter))),
+            }
 
-    let sortKeyFun = column.sortKeyFun;
-    if (sort.sortDesc) {
-        sortKeyFun = sortKeyFun && function (flow) {
-            const k = column.sortKeyFun(flow);
-            return _.isString(k) ? reverseString("" + k) : -k;
-        };
-    }
-    return sortKeyFun;
-}
+        case UPDATE_HIGHLIGHT:
+            return {
+                ...state,
+                highlight: action.highlight,
+            }
 
-export default function reducer(state = defaultState, action) {
-    switch (action.type) {
-        case UPDATE_FLOWS:
-            let all = reduceList(state.all, action)
+        case UPDATE_SORTER:
```

Comment:
We want to keep the state serializable, so we don't want to pass functions here. Can we please go back to the previous implementation?